src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -773,269 +774,292 @@ static HtmlEntities()
             private const int Count = 253;
 
             // maps entity strings => unicode chars
-            private static readonly LowLevelDictionary<string, char> s_lookupTable =
-                new LowLevelDictionary<string, char>(Count, StringComparer.Ordinal)
+            private static readonly LowLevelDictionary<ulong, char> s_lookupTable =
```

Comment:
Would it be faster if this is switched to regular `Dictionary` ?

`LowLevelDictionary` is left-over from the layering effort in the early .NET Core days. It is not as optimized as regular Dictionary.

Reply:
Outside of this change it seems we could aim to remove LLD from CoreFX entirely. It is only used here and in UriSyntax.cs.

Reply:
@jkotas worth me opening an issue to remove LLD from CoreFX? I am not sure how involved perf verification would have to be of such a change.

Reply:
https://github.com/dotnet/corefx/pull/29411

Reply:
I can confirm that replacing `LowLevelDictionary` with `Dictionary` does not affect the performance negatively. Here are some numbers (30000 calls to `HtmlDecode` per run):
#### `LowLevelDictionary` ####

 Run |     Mean |    Error |    StdDev |      Gen 0 | Allocated |
------- |---------:|---------:|----------:|-----------:|----------:|
   1 | 410.4 ms | 16.52 ms | 0.9337 ms | 59125.0000 | 177.61 MB |
   2 | 412.0 ms | 32.60 ms | 1.842 ms | 59125.0000 | 177.61 MB |
   3 | 411.4 ms | 36.08 ms | 2.039 ms | 59125.0000 | 177.61 MB |
   4 | 410.2 ms | 1.824 ms | 0.1030 ms | 59125.0000 | 177.61 MB |
   5 | 413.7 ms | 6.160 ms | 0.3480 ms | 59125.0000 | 177.61 MB |

#### Regular `Dictionary` ####

Run |     Mean |    Error |   StdDev |      Gen 0 | Allocated |
------- |---------:|---------:|---------:|-----------:|----------:|
   1 | 413.5 ms | 18.43 ms | 1.041 ms | 56375.0000 | 169.37 MB |
   2 | 406.3 ms | 18.16 ms | 1.026 ms | 56375.0000 | 169.37 MB |
   3 | 406.9 ms | 45.48 ms | 2.570 ms | 56375.0000 | 169.37 MB |
   4 | 410.8 ms | 84.31 ms | 4.763 ms | 56375.0000 | 169.37 MB |
   5 | 411.4 ms | 35.25 ms | 1.992 ms | 56375.0000 | 169.37 MB |